wowcig.lua

```diff
@@ -48,21 +49,32 @@ end)()
 
 local load, save, onexit, version = (function()
   local casc = require('casc')
+  local handle, err
   local url = 'http://us.patch.battle.net:1119/' .. args.product
   local bkey, cdn, ckey, version = casc.cdnbuild(url, 'us')
-  assert(bkey)
-  log('loading', version)
-  local handle, err = casc.open({
-    bkey = bkey,
-    cdn = cdn,
-    ckey = ckey,
-    cache = args.cache,
-    cacheFiles = true,
-    keys = encryptionKeys,
-    locale = casc.locale.US,
-    log = log,
-    zerofillEncryptedChunks = true,
-  })
+  if(args.localDir) then
+    handle, err = casc.open(args.localDir, {verifyHashes=false})
+    if handle then
+      log('using local directory instead of CDN.',args.localDir)
+    else
+      log('unable to open local directory',args.localDir)
+    end
+  end
+  if not handle then
```

Comment:
just fail if you specify --local and it doesn't work, if you specify --local that probably means you don't want to hit the CDN at all